### ISSUE TITLE
Jimple output for android is buggy

### ISSUE THREAD

**Issue by auppar** [soot-oss/soot#1465]
Ex:
MainActivity {
Runnable a = new Runnable() {
public void run() {

}
}

Runnable b = new Runnable() {
public void run() {

}
}
onCreate() {

Handler handler = new Handler();

handler.post(a);
handler.post(b);
}
}

//corresponding jimple representation for onCreate() 

onCreate() {

Runnable $u1;
Handler handler;
...
$u1 = this.a;
handler.post($u1); //1

$u1 = this.b;
handler.post($u1); //2

}

Notes : at tags 1 and 2 the same local variable of the type is used. Why are no separate local variables generated.

If i mimic the control flow in a normal java program i see that two local variables of the types runnable are created.

Some analysis result in false positives due to this representation.

I wish to know if there is any specific option to be set or is this a bug in the dexpler transformation of the android byte code to jimple.

**Comment by MarcMil** [comment]
Discussion on https://github.com/secure-software-engineering/FlowDroid/issues/280
TLDR: Not a bug. Just wrong expectations.
